modal-components/src/lib/components/modals/agreementModal/useAgreementCheck.tsx

```
@@ -0,0 +1,66 @@
+import { agreementContent } from "../../../types/modalTypes";
+
+export const useAgreementModal = ({
+  agreementContents,
+  modalSize,
+}: {
+  agreementContents: agreementContent[];
+  modalSize?: "small" | "medium" | "large";
+}) => {
+  const [checkboxes, setCheckboxes] = useState<Record<number, boolean>>(() =>
+    agreementContents.reduce(
+      (acc, _, index) => ({ ...acc, [index]: false }),
```

Biome (1.9.4)

[error] Avoid the use of spread (`...`) syntax on accumulators.

Spread syntax should be avoided on accumulators (like those in `.reduce`) because it causes a time complexity of `O(n^2)`.
Consider methods such as .splice or .push instead.

(lint/performance/noAccumulatingSpread)